File: database/src/main/java/com/pelisat/cesp/ceemsp/database/model/Can.java
package com.pelisat.cesp.ceemsp.database.model;

import com.pelisat.cesp.ceemsp.database.type.CanGeneroEnum;
import com.pelisat.cesp.ceemsp.database.type.CanOrigenEnum;
import com.pelisat.cesp.ceemsp.database.type.CanStatusEnum;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.Date;

@Entity
@Table(name = "CANES")
@Getter
@Setter
public class Can extends CommonModel {
    @Column(name = "NOMBRE", nullable = false)
    private String nombre;

    @Column(name = "EMPRESA", nullable = false)
    private Integer empresa;

    @Column(name = "GENERO", nullable = false)
    @Enumerated(EnumType.STRING)
    private CanGeneroEnum genero;

    @Column(name = "RAZA", nullable = false)
    private Integer raza;

    @Column(name = "RAZA_OTRO")
    private String razaOtro;

    @Column(name = "DOMICILIO_ASIGNADO", nullable = false)
    private Integer domicilioAsignado;

    @Column(name = "FECHA_INGRESO", nullable = false)
    private LocalDate fechaIngreso;

    @Column(name = "EDAD", nullable = false)
    private Integer edad;

    @Column(name = "PESO", nullable = false)
    private BigDecimal peso;

    @Column(name = "DESCRIPCION")
    private String descripcion;

    @Column(name = "ORIGEN", nullable = false)
    @Enumerated(EnumType.STRING)
    private CanOrigenEnum origen;

    @Column(name = "STATUS", nullable = false)
    @Enumerated(EnumType.STRING)
    private CanStatusEnum status;

    @Column(name = "CHIP", nullable = false)
    private Boolean chip;

    @Column(name = "TATUAJE", nullable = false)
    private Boolean tatuaje;

    @Column(name = "RAZON_SOCIAL")
    private String razonSocial;

    @Column(name = "FECHA_INICIO")
    private LocalDate fechaInicio;

    @Column(name = "FECHA_FIN")
    private LocalDate fechaFin;

    @Column(name = "ELEMENTO_ASIGNADO")
    private Integer elementoAsignado;

    @Column(name = "CLIENTE_ASIGNADO")
    private Integer clienteAsignado;

    @Column(name = "DOMICILIO_CLIENTE_ASIGNADO")
    private Integer domicilioClienteAsignado;

    @Column(name = "MOTIVOS")
    private String motivos;

    @Column(name = "MOTIVO_BAJA")
    private String motivoBaja;

    @Column(name = "OBSERVACIONES_BAJA")
    private String observacionesBaja;

    @Column(name = "DOCUMENTO_FUNDATORIO_BAJA")
    private String documentoFundatorioBaja;

    @Column(name = "FECHA_BAJA")
    private LocalDate fechaBaja;

    @Column(name = "FOTOGRAFIA_CAPTURADA")
    private boolean fotografiaCapturada;

    @Column(name = "ADIESTRAMIENTO_CAPTURADO")
    private boolean adiestramientoCapturado;

    @Column(name = "VACUNACION_CAPTURADA")
    private boolean vacunacionCapturada;

    @Column(name = "CONSTANCIA_CAPTURADA")
    private boolean constanciaCapturada;
}

File: ceemsp-frontend/src/app/_models/Can.ts
import {Deserializable} from "./Deserializable";
import CanRaza from "./CanRaza";
import EmpresaDomicilio from "./EmpresaDomicilio";
import Persona from "./Persona";
import Cliente from "./Cliente";
import ClienteDomicilio from "./ClienteDomicilio";
import CanCartillaVacunacion from "./CanCartillaVacunacion";
import CanConstanciaSalud from "./CanConstanciaSalud";
import CanAdiestramiento from "./CanAdiestramiento";

export default class Can implements Deserializable {
  id: number;
  uuid: string;
  nombre: string;
  genero: string;
  raza: CanRaza;
  domicilioAsignado: EmpresaDomicilio;
  fechaIngreso: string;
  edad: number;
  peso: number;
  descripcion: string;
  origen: string;
  status: string;
  chip: boolean;
  tatuaje: boolean;
  razonSocial: string;
  fechaInicio: string;
  fechaFin: string;
  elementoAsignado: Persona;
  clienteAsignado: Cliente;
  clienteDomicilio: ClienteDomicilio;
  motivos: string;

  cartillasVacunacion: CanCartillaVacunacion[];
  constanciasSalud: CanConstanciaSalud[];
  adiestramientos: CanAdiestramiento[];

  deserialize(input: any): this {
    Object.assign(this, input);
    return this;
  }

}
